### PR TITLE
Home Assistant presence

### PR DESCRIPTION
**Pull request by Project-Piffa**
Home Assistant presence 1.0.0

I created the pull request for the local domain for home assistant
image
image
image
image

### CODE REVIEW

File: websites/H/HomeAssistant/dist/metadata.json [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  },
+  "url": "homeassistant.local",
+  "description": {
+      "en": "Open source home automation that puts local control and privacy first. Powered by a worldwide community of tinkerers and DIY enthusiasts. Perfect to run on a Raspberry Pi or a local server.",
+      "it": "Domotica open source che mette al primo posto il controllo locale e la privacy. Alimentato da una comunità mondiale di armeggiatori e appassionati di fai da te. Perfetto per funzionare su un Raspberry Pi o un server locale."
+  },
+  "service": "Home_Assistant",
+  "version": "1.0.0",
+  "logo": "https://upload.wikimedia.org/wikipedia/commons/6/6e/Home_Assistant_Logo.svg",
+  "thumbnail": "https://www.home-assistant.io/images/default-social.png",
```
Comment (Bas950) [inline websites/H/HomeAssistant/dist/metadata.json:15]: upload to imgur

File: websites/H/HomeAssistant/dist/metadata.json [hunk 2]
```diff
@@ -0,0 +1,25 @@
+{
+  "$schema": "https://schemas.premid.app/metadata/1.0",
+  "author": {
+      "name": "Project_Piffa",
+      "id": "525634922321412096"
+  },
+  "url": "homeassistant.local",
```
Comment (Bas950) [inline websites/H/HomeAssistant/dist/metadata.json:7]: is that even a valid domain?

File: websites/H/HomeAssistant/dist/metadata.json [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+  "description": {
+      "en": "Open source home automation that puts local control and privacy first. Powered by a worldwide community of tinkerers and DIY enthusiasts. Perfect to run on a Raspberry Pi or a local server.",
+      "it": "Domotica open source che mette al primo posto il controllo locale e la privacy. Alimentato da una comunità mondiale di armeggiatori e appassionati di fai da te. Perfetto per funzionare su un Raspberry Pi o un server locale."
+  },
+  "service": "Home_Assistant",
```
Comment (Bas950) [inline websites/H/HomeAssistant/dist/metadata.json:12]: Service name and folder name must be the same and you can use spaces.

File: websites/H/HomeAssistant/dist/metadata.json [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+  "version": "1.0.0",
+  "logo": "https://upload.wikimedia.org/wikipedia/commons/6/6e/Home_Assistant_Logo.svg",
+  "thumbnail": "https://www.home-assistant.io/images/default-social.png",
+  "color": "#3DB4EB",
+  "tags": [
+      "IOT",
+      "HomeAssistant",
+      "Raspberry",
+      "HomeAutomation",
+      "Home"
```
Comment (Bas950) [inline websites/H/HomeAssistant/dist/metadata.json:22]: tags cant have capped letters.

File: websites/H/HomeAssistant/presence.ts [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+  function homeassistant(nome: string, imagetext: string, details: string) {
+    presenceData.smallImageKey = nome;   // SmallImageKey Function
+    presenceData.smallImageText = imagetext;   // smallImageText Function
+    presenceData.details = details;    // Details Function
+  }
+
+  //#region Main
+  if (document.location.pathname.includes("/lovelace")) {
+    homeassistant("home_assistant_logo,", "Home Assistant", "Home")
+  }
```
Comment (Bas950) [inline websites/H/HomeAssistant/presence.ts:20]: your key and smallimagetext is always the same... so just set it at the top...

for the others maybe do like Viewing the homepage etc. not just Home

**Review by Bas950 — CHANGES_REQUESTED** [review]
Upload proof of it working

**Comment by Project-Piffa** [comment]
Hey, i fixed the problem, anyway yes homeassistant.local is a valid **local**  domain so is accessible only in your LAN if you have installed home assistant in a server :)

ha2
ha1
ha5
ha4
ha3

File: websites/H/Home Assistant/presence.ts [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+  const presenceData: PresenceData = {
+    largeImageKey: "home_assistant_logo" //banner
+  }; 
+  presenceData.startTimestamp = browsingStamp;
+  
+
+  function homeassistant(state: string) {
+    presenceData.details = "Viewing";
+    presenceData.state = state;
+  }
```
Comment (ririxidev) [inline websites/H/Home Assistant/presence.ts:15]: It isn't good option to do it...

File: websites/H/Home Assistant/presence.ts [hunk 7]
```diff
@@ -0,0 +1,131 @@
+const presence = new Presence({
+    clientId: "812627990382772224"
+}), browsingStamp = Math.floor(Date.now() / 1000);
+  
+presence.on("UpdateData", async () => {
+  const presenceData: PresenceData = {
+    largeImageKey: "home_assistant_logo" //banner
+  }; 
+  presenceData.startTimestamp = browsingStamp;
```
Comment (ririxidev) [inline websites/H/Home Assistant/presence.ts:9]: just, add startTimestamp to presenceData object?

**Review by ririxidev — CHANGES_REQUESTED** [review]

File: websites/H/Home Assistant/presence.ts (hunk 6 above)
Comment (diskxo) [reply websites/H/Home Assistant/presence.ts:15]: How can we change this? Presencedata details is always "Viewing"

File: websites/H/Home Assistant/presence.ts (hunk 7 above)
Comment (diskxo) [reply websites/H/Home Assistant/presence.ts:9]: What do you mean? We have to insert startTimestamp into const presenceData?

**Comment by Project-Piffa** [comment]
Hey, I had to reload the old code because with the changes made it gave an error, I only know this way to make startTimestamp and always appear "viewing".
If there are other methods tell me it please :)

File: websites/H/Home Assistant/presence.ts [hunk 8]
```diff
@@ -0,0 +1,131 @@
+const presence = new Presence({
+    clientId: "812627990382772224"
+}), browsingStamp = Math.floor(Date.now() / 1000);
+  
+presence.on("UpdateData", async () => {
+  const presenceData: PresenceData = {
+    largeImageKey: "home_assistant_logo" //banner
+  }; 
+  
+  presenceData.startTimestamp = browsingStamp; // Start browsingStamp
```
Comment (ririxidev) [inline websites/H/Home Assistant/presence.ts:10]:
```typescript
const presenceData: PresenceData = {
    largeImageKey: "home_assistant_logo" //banner
    startTimestamp: browsingStamp; // Start browsingStamp
};
``` 
change to this... guys...

File: websites/H/Home Assistant/presence.ts (hunk 6 above)
Comment (ririxidev) [reply websites/H/Home Assistant/presence.ts:15]: It doesn't always mean that it must be "Viewing" everytime??

**Review by ririxidev — CHANGES_REQUESTED** [review]
fix deepscan issues too

**Comment by diskxo** [comment]
@ririxidev yea, the presencedata.details must aways set to "viewing"

**Comment by Project-Piffa** [comment]
Hey, thank you for the advice, I hope that I fixed the problems, anyway yes the presencedata.details must always set to "viewing".
In DeepScan I see one issues but isn't in my directory.

File: websites/H/Home Assistant/presence.ts [hunk 9]
```diff
@@ -0,0 +1,130 @@
+const presence = new Presence({
+    clientId: "812627990382772224"
+}), browsingStamp = Math.floor(Date.now() / 1000);
+  
+presence.on("UpdateData", async () => {
+  const presenceData: PresenceData = {
+    largeImageKey: "home_assistant_logo", //banner
+    startTimestamp: browsingStamp
+  };
```
Comment (Bas950) [inline websites/H/Home Assistant/presence.ts:9]: just set the details in here...;.

**Review by Bas950 — CHANGES_REQUESTED** [review]

File: websites/H/Home Assistant/presence.ts (hunk 9 above)
Comment (ririxidev) [reply websites/H/Home Assistant/presence.ts:9]: and it means to delete this function after doing it thanks.

**Comment by Project-Piffa** [comment]
Hey, I fixed the problem  and i romoved the funciton :)

File: websites/H/Home Assistant/presence.ts [hunk 10]
```diff
@@ -5,119 +5,115 @@ const presence = new Presence({
...
 
   else if (document.location.pathname.includes("/profile")) {
-    homeassistant("Profile");
+    presenceData.state = "Profile";
   }
   else{
-    homeassistant("Navigating...");
+    presenceData.state = "Navigating...";
     presenceData.details = "";
   }
```
Comment (Bas950) [inline websites/H/Home Assistant/presence.ts]: use just set the details to navigating and dont set the state.

**Review by Bas950 — CHANGES_REQUESTED** [review]

**Comment by Project-Piffa** [comment]
Hey, I setted the details to navigating and I removed the state

**Review by Bas950 — APPROVED** [review]

**Comment by ririxidev** [comment]
Hey, unfortunately, we will reject this presence. Why this happened? This presence is just low quality for us, since presence is only showing on your subpage you are and it isn't that great thing to accept it as "good" quality presence. If site doesn't allow you to get things to show, it's unfortunately too. Sorry then, maybe next time! 😃

**Pull request closed without merging**